### DISCUSSION TITLE
Product design and UI/UX Support

### DISCUSSION THREAD

**Discussion by fivenp** [akuity/kargo#917] [Ideas]
Hey folks. 

I really love what you're doing over here and I'm super curious if you'd be up for some Design support? :) 

After trying the UI out myself I couldn't resist to fiddle around with some Ideas I had in my mind. 

![image](https://github.com/akuity/kargo/assets/132332/45bf2649-5bfa-469e-a13a-aac5880eaf57)

My main thoughts were: 

1. I really feel Kargo would benefit from a slick, modern and shiny look (I think it's time for K8s tools to level up and reach the same wow factor as oder modern day tools out there :))
2. From my POV Kargo's MAIN thing is visualising the current deployment status – but also giving users A LOT of deployment insights at a glance (Hence the recent deployments section) – what was deployed when why how and what/who triggered it
3. I did realise that promoting a freight to a stage or selecting a freight for a stage are rather clunky to use right now. The functionality requires the user to select a stage first (on the bottom) and then assign a freight to it (at the top) (which is counter-intuitive to typical UX left to right / top to bottom flows) - that said - I think very basic context menus on both freights and stages can solves this easily. Alternative would be drag'n drop support from freight to stage might also be pretty nice.....

<img width="458" alt="image" src="https://github.com/akuity/kargo/assets/132332/531bbb4a-8caa-49a1-979d-10b1d31fa03d">
<img width="495" alt="image" src="https://github.com/akuity/kargo/assets/132332/a9e11096-0a02-4b07-865c-9e8ce55127e4">

Sorry for so many thoughts at once upfront... But I think the main question here is – would you be up for some Design collabos? If so – how should we go about it? :)

**Comment by christianh814** [comment]
cc @rbreeze

**Comment by krancour** [comment]
@fivenp, first let me say these are gorgeous! You've got a lot of great ideas here.

The maintainers have spoken internally and we all agree that community contributions are not only welcome, but encouraged -- but in form of more narrowly-scoped, _incremental_ improvements. Changes of this magnitude are not on the table at this time.

**Comment by fivenp** [comment]
@krancour thx for the kind words.... 

I fully agree on the incremental / iterative approach... Obviously a whole revamp of the dashboard AND a huge increase in functionality at the same time does make 0 sense tbh :) 

I do think however the improvements / ideas COULD be implemented in a very iterative way to be fair (in case the project even WANTS to go into the proposed direction)

For example

Step 1: Keeping the current style but adding context menus to Freights / Stages
Step 2: Keep the current UX and content but adapt the style
Step 3: Do more user research on what people want to do/see at a glance and run the whole product/design cycle to implement those things (e.g. proposed deployment overviews – which might potentially be totally wrong :)) 

Here's an example of how Step 2 could look like (maybe it can be scoped down even more): 
![image](https://github.com/akuity/kargo/assets/132332/b0614302-942e-4708-8252-e511a7a68e91)


**However I do have some proposals – which I'd be glad if you could add your thoughts**
* If you're short on resources regarding the style changes – I could theoretically prepare a dashboard with no functionality but with all the components for you folks? 
* Going forward I'd love to be part of your feature creation process (including design at an early stage) – just in case you are up for it at all :) (we can handle this through e.g. GH Issues with a corresponding tag)

**Reply by krancour** [reply]
@fivenp please try to understand this is an MVP. A corollary to that is that we have a team of dedicated engineers _and designers_ who are actively iterating on this.

We really _do_ need to get a public roadmap out there so that it's more obvious that this is NOT the end state. We could stand to be more transparent about current areas of focus.

All of this said, a large dump of "here's what _I_ think you guys should do" is not feeling helpful to us. To be honest, it feels more like being carjacked.

We continue to value everyone's input, but please open narrowly focused issues, please understand that the maintainers won't accept _every_ idea, and please understand that when accepted ideas don't align with _current_ priorities, the best chance of seeing them come to fruition will lie in opening narrowly-scoped PRs yourself.

**Comment by fivenp** [comment]
I am very sorry if it came across as carjacking or "here is what you should do" - that was def. not the intention after all.

I was rather trying to understand how to be able to contribute.

Anyway. Thanks for taking the time and this thorough answer 🙂

**Comment by Fazna003** [comment]
I’ve been playing around with Kargo’s UI and had a few ideas. First off, I think Kargo could use a modern and stylish look. It's time for Kubernetes tools to have that 'wow' factor!

The core of Kargo is showing deployment status and providing quick insights. I added a recent deployments section for users to see what was deployed, when, why, and by whom.

One thing I noticed: assigning freight to a stage feels clunky. I suggest adding basic context menus for freights and stages or allowing drag-and-drop functionality. This way, the process becomes more intuitive, following the usual left-to-right, top-to-bottom flow.

By the way, if you’re interested, my [UI/UX design company ](https://www.aufaitux.com/)has experience in creating seamless and user-friendly interfaces. We could collaborate to give Kargo the modern edge it deserves.

**Reply by krancour** [reply]
@Fazna003 no thank you. It should be clear from previous discussion that we have our own designers diligently and continuously iterating on this.
